mp/game/bin/base.fgd

```diff
@@ -0,0 +1,6582 @@
+	input SetTonemapRate(float) : "Set the rate for autoexposure adjustment."
+]
+
+@PointClass base(Targetname, Parentname) sweptplayerhull() = func_useableladder : 
+	"A Half-Life 2 ladder. Handles player auto mount/unmount, as well as +use to get onto the ladder. \n\n" +
+	"See also 'info_ladder_dismount', used to specify ladder auto-dismount points.\n\n" +
+	"Note: This entity is non-functional in Counter-Strike: Source. Use func_ladder instead."
+[
+	spawnflags(flags) =
+	[
+		1: "Fake Ladder" : 0
+	]
+
+	point0(vector) : "Start" : : "Ladder end point."
+	point1(vector) : "End" : : "Ladder end point."
+
+	StartDisabled(choices) : "Start Disabled" : 0 =
+	[
+		0 : "No"
+		1 : "Yes"
+	]
+
+	// Inputs
+	input Enable(void) : "Enable this ladder."
+	input Disable(void) : "Disable this ladder."
+
+	output OnPlayerGotOnLadder(void) : "Fired whenever a player gets on this ladder."
+	output OnPlayerGotOffLadder(void) : "Fired whenever a player gets off this ladder."
+	ladderSurfaceProperties(string) : "Surface properties (optional)"
+]
+
+@PointClass base(Targetname, Parentname, Angles) size( -16 -16 0, 16 16 72 ) color(127 127 127) = func_ladderendpoint : 
+	"An entity used to specify the endpoints of a ladder. This entity is functional, but has been replaced by the " + 
+	"easier-to-use func_useableladder entity. Left in only for backwards-compatibility!\n\n" +
+	"To be valid, a full sized player hull traced between the start and end points must not be obstructed at level " + 
+	"activation time. The angle determines in which direction the player leaves the ladder if the player presses the " +
+	"+jump button.\n\n" +
+	"Note: This entity is non-functional in Counter-Strike: Source. In CS:S, use func_ladder instead."
+[
+	target(target_destination) : "Other" : : "A ladder goes between any two func_ladderendpoints pointing at each other."
+	// TODO:  Allow individual ladder end points to be enabled, disabled
+	// TODO:  Allow ladder to specify movement speed or a speed scale while on ladder
+	// TODO:  Allow specifying radius within with you must be in order to +use to get on the ladder
+]
+
+@PointClass base(Parentname) size( -16 -16 0, 16 16 4 ) color(255 128 255)= info_ladder_dismount : 
+	"An entity to handle endpoints for multiple ladders that are too close to each other."
+[
+	target(target_destination) : "LadderName" : : "If multiple ladders are near multiple endpoints, use this to stop them from interfering with each other."
+]
+
+@SolidClass base(Targetname) color(0 128 255) = func_areaportalwindow : 
+	"An entity that can be used to optimize the visibility in a map. If you seal off an area with them, when the viewer moves the specified distance away from them, they will go opaque and the parts inside the area will not be drawn. The 'target' brush model should enclose the func_areaportal window so no parts of it are culled by the window. If you use the optional foreground brush model, then it should enclose the 'target' brush mode
```

Comment:
We do override `env_cubemap` in our momentum.fgd file for parallax corrected cubemaps. You can actually make another commit that moves the overridden cubemap back into here, to keep our momentum.fgd file to be purely momentum-specific entities.